services/registration/src/routes/application.ts

```diff
@@ -41,18 +41,27 @@ applicationRouter.route("/").post(
   asyncHandler(async (req, res) => {
     await validateApplicationData(req.body.applicationBranch, req.body.applicationData);
 
-    const newApplication = await ApplicationModel.create({
-      userId: req.body.userId,
-      hexathon: req.body.hexathon,
-      applicationBranch: req.body.applicationBranch,
-      applicationData: req.body.applicationData,
-      applicationStartTime: req.body.applicationStartTime,
-      applicationSubmitTime: req.body.applicationSubmitTime,
-      confirmationBranch: req.body.confirmationBranch,
-      confirmationData: req.body.confirmationData,
-      confirmationStartTime: req.body.confirmationStartTime,
-      confirmationSubmitTime: req.body.confirmationSubmitTime,
-    });
+    let newApplication;
+    if (req.body.applicationStatus === "SUBMIT") {
+      newApplication = await ApplicationModel.create({
+        userId: req.body.userId,
+        hexathon: req.body.hexathon,
+        applicationBranch: req.body.applicationBranch,
+        applicationData: req.body.applicationData,
+        applicationStartTime: req.body.applicationStartTime,
+        applicationSubmitTime: req.body.applicationSubmitTime,
+        applicationStatus: req.body.applicationStatus,
+      });
+    } else {
+      newApplication = await ApplicationModel.create({
+        userId: req.body.userId,
+        hexathon: req.body.hexathon,
+        applicationBranch: req.body.applicationBranch,
+        applicationData: req.body.applicationData,
+        applicationStartTime: req.body.applicationStartTime,
+        applicationStatus: req.body.applicationStatus,
+      });
```

Comment:
I feel like you could simplify this a little into one create, and then based on the `applicationStatus` update the submit time or not. Additionally, the userId won't be passed in via the `req.body`, you'll need to read that from `req.user`